core/standalone/src/main/scala/org/apache/openwhisk/standalone/StandaloneOpenWhisk.scala

```diff
@@ -63,6 +64,25 @@ class Conf(arguments: Seq[String]) extends ScallopConf(arguments) {
   val apiGwPort = opt[Int](descr = "Api Gateway Port", default = Some(3234), noshort = true)
   val dataDir = opt[File](descr = "Directory used for storage", default = Some(StandaloneOpenWhisk.defaultWorkDir))
 
+  val kafka = opt[Boolean](descr = "Enable embedded Kafka support", noshort = true)
+  val kafkaUi = opt[Boolean](descr = "Enable Kafka UI", noshort = true)
+
+  val kafkaPort = opt[Int](
+    descr = "Kafka port. If not specified then 9092 or some random free port (if 9092 is busy) would be used",
+    noshort = true,
+    required = false)
```

Comment:
Would it make sense to add a `default = Some(9092)` here?

Reply:
This is done on purpose. Port 9092 is more like a "preferred" port. If no explicit option is provided then system would try to start Kafka by default at 9092. However if that port is busy then a random port would be selected. 

This is done to ensure that Standalone OpenWhisk needs minimum required free port (for now only 3233). Rest all services can be started at any port and then dependent service would be configured as per randomly selected port

If a user explicitly provides a port like `--kafka-port 9010` then Kafka would be started at that port. If the port is busy then Kafka would fail to start. This support is mostly meant for test which would like to know where Kafka would start

Reply:
I understand. Background of my proposal was that there is some duplication of the default ports. We have the port number 9092 for Kafka here AND in `startKafka()`. Wouldn't it make sense to define a `val defaultKafkaPort = 9092` and use it to build the description string and in `startKafka()`?

Reply:
👍 Done that now